Model/Order/PlaceOrder.php

```diff
@@ -465,74 +343,23 @@ private function getValidationErrorResponse(string $message): ResultInterface
         );
     }
 
-    /**
-     * @return array{
-     *      data?: array{
-     *          total: int,
-     *          transactions: array{
-     *              gateway: string,
-     *              payment_id: string,
-     *              amount: int,
-     *              transaction_id: string,
-     *              currency: string,
-     *              step: string,
-     *              status: 'success'|'failed'|'',
-     *              tender_type: string,
-     *              tender_details: array{
-     *                  brand: string,
-     *                  last_four: string,
-     *                  bin: string,
-     *                  expiration: string
-     *              },
-     *              gateway_response_data: string[]
-     *          }[]
-     *      },
-     *      errors?: array{
-     *          message: string,
-     *          type: string,
-     *          field: string,
-     *          severity: string,
-     *          sub_type: string,
-     *          code?: string,
-     *          transactions?: array{
-     *              gateway: string,
-     *              payment_id: string,
-     *              amount: int,
-     *              transaction_id: string,
-     *              currency: string,
-     *              step: string,
-     *              status: 'success'|'failed'|'',
-     *              tender_type: string,
-     *              tender_details: array{
-     *                  brand: string,
-     *                  last_four: string,
-     *                  bin: string,
-     *                  expiration: string
-     *              },
-     *              gateway_response_data: string[]
-     *          }[]
-     *      }[]
-     *  }
-     * @throws Exception
-     */
     private function getAuthorizedPayments(string $publicOrderId, int $websiteId): array
     {
         $url = sprintf('checkout/orders/{{shopId}}/%s/payments/auth/full', $publicOrderId);
         $result = $this->client->post($websiteId, $url, []);
         $errors = $result->getErrors();
-
-        if (array_key_exists('errors', $errors) && count($errors) > 0) {
-            $errors = [
-                [
-                    'code' => $result->getStatus(),
-                    'type' => 'server.bold_checkout_api_error',
-                    'message' => $errors[0],
-                    'transactions' => []
-                ]
-            ];
+        $response = $result->getBody();
+
+        if (count($errors) > 0) {
+            $error = $errors[0];
+            $errorMessage = isset($error['message']) ? $error['message'] : 'Unknown error occurred';
+            $errorType = isset($error['type']) ? $error['type'] : null;
+            return ['success' => false, 'error' => $errorMessage, 'type' => $errorType];
+        } else if (!isset($response['data']) || $response['data'] === null) {
+            return ['success' => false, 'error' => 'No data found'];
```

Comment:
[Preference] This reads cleaner to me and has error handling built in, previous the throwing and catching of errors was taking longer and I've tested both success and failure cases. 
Let me know if there's any thoughts / preferences here.

Reply:
IMHO, the previous logic was clearer. If we want to go this route, though, the `else if` statement could just be made into its own `if` statement (the `else` is not necessary in other words). I would also add a space between the assignments and return statement to make it more readable.

Reply:
This reads cleaner to me, but always happy to get another opinion. I'm fine to switch them to all guard statements, but I think that's personal preference.